src/app/course/course-routing.module.ts

```diff
@@ -0,0 +1,57 @@
+import {AuthGuard} from "@app/_helpers/auth.guard";
+import {NgModule} from "@angular/core";
+import {RouterModule} from "@angular/router";
+import {CourseListComponent} from "@app/course/course-list/course-list.component";
+import {CourseRegisterComponent} from "@app/course/course-registration/course-register.component";
+import {CourseComponent} from "@app/course/course.component";
+import {CourseEventCreateEditComponent} from "@app/course/course-event-create/course-event-create-edit.component";
+import {CourseQuestionSnippetComponent} from "@app/course/course-question-snippet/course-question-snippet.component";
+import {ProblemViewComponent} from "@app/problems/problem-view/problem-view.component";
+
+const routes = [
+    {
+        path: '',
+        component: CourseListComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'register/:courseId',
+        component: CourseRegisterComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: 'view/:courseId',
+        component: CourseComponent,
+        canActivate: [AuthGuard]
+    },
+    {
+        path: ':courseId/new-event',
+        component: CourseEventCreateEditComponent,
```

Comment:
Please organize these paths. All the ones start with `:courseId` should be nested.